### PR TITLE
Python scripts

### PR DESCRIPTION
**Pull request by albertodonato**
Convert the mirror and signing helper scripts to python, so that logic can be tested and reused more easily.

### CODE REVIEW

**Comment by albertodonato** [comment]
[Move scripts to python](https://trello.com/c/jKVRf9H0/232-move-scripts-to-python)

File: lib/charms/archive_auth_mirror/setup.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def install_resources(root_dir=None):
         shutil.copy(resource_path, str(paths['bin']))
     # install crontab
     shutil.copy('resources/crontab', str(paths['cron']))
+    # symlink the lib libary to make it available to scripts too
+    os.symlink(os.path.join(os.getcwd(), 'lib'), str(paths['bin'] / 'lib'))
```
Comment (sparkiegeek) [inline lib/charms/archive_auth_mirror/setup.py]: :thinking: `(paths['bin'] / 'lib').symlink_to(Path.cwd() / 'lib')` :question:

File: unit_tests/test_reprepro_sign_helper.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    inline_sign,
+    detach_sign,
+)
+
+
+class GPGCallsTest(TestCase):
+
+    @mock.patch('subprocess.check_call')
+    def test_gpg(self, mock_check_call):
+        """The gpg function calls gpg."""
```
Comment (tealeg) [inline unit_tests/test_reprepro_sign_helper.py]: I think we can do a better docstring than this.  How about "The gpg function invokes the system gpg binary, with appropriate arguments." ?

Comment (albertodonato) [inline unit_tests/test_reprepro_sign_helper.py]: done, thanks

File: lib/charms/archive_auth_mirror/setup.py (hunk 1 above)
Comment (albertodonato) [inline lib/charms/archive_auth_mirror/setup.py]: Nice, done

**Pull request merged**